docs/tutorial/composition.md

```diff
@@ -0,0 +1,221 @@
+# Composition
+
+`concrete-python` supports circuit __composition__, which allows the output of a circuit execution to be used directly as an input without decryption. We can execute the circuit as many time as we want by forwarding outputs without decrypting intermediate values. This feature enables a new range of applications, including support for control flow in pure (cleartext) python.
+
+Here is a first simple example that uses composition to implement a simple counter in FHE:
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"counter": "encrypted"})
+def increment(counter):
+   return (counter + 1) % 100
+
+print("Compiling `increment` function")
+increment_fhe = increment.compile(list(range(0, 100)), composable=True)
+
+print("Generating keyset ...")
+increment_fhe.keygen()
+
+print("Encrypting the initial counter value")
+counter = 0
+counter_enc = increment_fhe.encrypt(counter)
+
+print(f"| iteration || decrypted | cleartext |")
+for i in range(10):
+    counter_enc = increment_fhe.run(counter_enc)
+    counter = increment(counter)
+
+    # For demo purpose; no decryption is needed.
+    counter_dec = increment_fhe.decrypt(counter_enc)
+    print(f"|     {i}     || {counter_dec:<9} | {counter:<9} |")
+```
+
+Note the use of the `composable` flag in the `compile` call. It instructs the compiler to ensure the circuit can be called on its own outputs (see [Limitations section](#limitations) for more details). Executing this script should give the following output:
+
+```shell
+Compiling `increment` function
+Generating keyset ...
+Encrypting the initial counter value
+| iteration || decrypted | cleartext |
+|     0     || 1         | 1         |
+|     1     || 2         | 2         |
+|     2     || 3         | 3         |
+|     3     || 4         | 4         |
+|     4     || 5         | 5         |
+|     5     || 6         | 6         |
+|     6     || 7         | 7         |
+|     7     || 8         | 8         |
+|     8     || 9         | 9         |
+|     9     || 10        | 10        |
+```
+
+## Multi inputs, multi outputs
+
+Composition is not limited to 1-to-1 circuits, it can also be used with circuits with multiple inputs and multiple outputs. Here is an example that computes the 10 first elements of the Fibonacci sequence in FHE:
+
+```python
+from concrete import fhe
+
+def identity(x):
```

Comment:
maybe 'identity' could be name using its purpose
e.g. noise_reset / refresh
also it would be a perfect decorator to include
so people can say
```python
@fhe.compiler(....)
@noise reset
def fib(...)
```
```